sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -287,53 +291,57 @@
      * {@inheritDoc}
      */
     @Override
-    public Mono<Long> schedule(ServiceBusMessage message, OffsetDateTime scheduledEnqueueTime, int maxLinkSize,
-        String associatedLinkName, ServiceBusTransactionContext transactionContext) {
-        message.setScheduledEnqueueTime(scheduledEnqueueTime);
-
-        return isAuthorized(OPERATION_SCHEDULE_MESSAGE).then(createChannel.flatMap(channel -> {
-            // Serialize the request.
-            final Message amqpMessage = messageSerializer.serialize(message);
-
-            // The maxsize allowed logic is from ReactorSender, this logic should be kept in sync.
-            final int payloadSize = messageSerializer.getSize(amqpMessage);
-            final int allocationSize =
-                Math.min(payloadSize + ManagementConstants.MAX_MESSAGING_AMQP_HEADER_SIZE_BYTES, maxLinkSize);
-            final byte[] bytes = new byte[allocationSize];
-
-            int encodedSize;
-            try {
-                encodedSize = amqpMessage.encode(bytes, 0, allocationSize);
-            } catch (BufferOverflowException exception) {
-                final String errorMessage = String.format(
-                    "Error sending. Size of the payload exceeded maximum message size: %s kb", maxLinkSize / 1024);
-                final AmqpErrorContext errorContext = channel.getErrorContext();
-
-                return monoError(logger, Exceptions.propagate(new AmqpException(false,
-                    AmqpErrorCondition.LINK_PAYLOAD_SIZE_EXCEEDED, errorMessage, exception, errorContext)));
-            }
+    public Flux<Long> schedule(List<ServiceBusMessage> messages, OffsetDateTime scheduledEnqueueTime,
```

Comment:
We're basically creating our own message batch here. Why not just have a schedule that takes MessageBatch? Then we don't need the maxLinkSize anymore.

Reply:
messageBatch.getMessages() is package private and not accessible to `ManagementChannel` . And I did not want to make it public, Thus I am not sending `MessageBatch` here.

Reply:
I'd use an EntityHelper in this case because we're duplicating unnecessary mandatory logic here.